packs/data/feats.db/perfect-resistance.json

```diff
@@ -25,7 +25,35 @@
                 }
             ]
         },
-        "rules": [],
+        "rules": [
+            {
+                "key": "ActiveEffectLike",
+                "mode": "override",
+                "path": "flags.pf2e.houseOfPerfection.element",
+                "priority": 1,
+                "value": ""
+            },
+            {
+                "domain": "all",
+                "key": "RollOption",
+                "label": "PF2E.SpecificRule.HouseOfPerfection.PerfectResistance.Label",
+                "option": "used-sop-ki-spell",
+                "toggleable": true
+            },
+            {
+                "key": "Resistance",
+                "type": "{actor|flags.pf2e.houseOfPerfection.element}",
```

Comment:
You don't have resistance to the element, you have resistance to effects with that elemental trait. This one is going to be harder to pull off.

Reply:
Oops, wrong github account :D

Reply:
I confirmed with @stwlam that this support is in, and testing it shows that it works.

For example, using Hydraulic Push against Unbreaking Waves resists the bludgeoning damage as it has the water trait.